internal/manager/manager.go

```diff
@@ -50,22 +53,37 @@ type Config struct {
 }
 
 func New(ctx context.Context, cfg *Config) *Manager {
-	rc := redis.NewClient(&redis.Options{
-		Addr: cfg.RedisAddr,
-		OnConnect: func(c *redis.Conn) error {
-			if cfg.PodID != "" {
-				c.ClientSetName(cfg.Name + "-" + cfg.PodID)
-			} else {
-				c.ClientSetName(cfg.Name)
+	multiRedisEnv := os.Getenv("multi_redis")
+	var multiRedisAddresses []string
+	var rc *redis.Client
+	var rdbClients []*redis.Client
+	var err error
+
+	if multiRedisEnv != "" {
+		err = json.Unmarshal([]byte(multiRedisEnv), &multiRedisAddresses)
+		if err != nil {
+			cfg.Logger.Fatal(ctx, "invalid multi_redis", slog.Error(err))
+		}
+		for _, addr := range multiRedisAddresses {
+			var mrc *redis.Client
+			mrc, err = createRedisClient(addr, cfg.Name, cfg.PodID)
+			if err != nil {
+				// It is not fatal if one multiRedis client did not connect.
+				cfg.Logger.Warn(ctx, "createRedisClient", slog.Error(err))
+				continue
 			}
+			rdbClients = append(rdbClients, mrc)
+		}
+	} else {
+		rc, err = createRedisClient(cfg.RedisAddr, cfg.Name, cfg.PodID)
+		if err != nil {
+			cfg.Logger.Fatal(ctx, "createRedisClient", slog.Error(err))
+		}
+	}
 
-			return nil
-		},
-	})
-
-	_, err := rc.Ping().Result()
-	if err != nil {
-		cfg.Logger.Fatal(ctx, "failed to ping redis", slog.Error(err))
+	// No multi redis clients were connected, or all failed to connect.
+	if multiRedisEnv != "" && len(rdbClients) == 0 {
+		cfg.Logger.Fatal(ctx, "multiRedisEnv is set, but all redis clients failed to connect.")
 	}
```

Comment:
oops, not a problem but maybe can move this inside the first `if multiRedisEnv != ""` statement, right after the for loop?